### PR TITLE
Address WoL issue 25793

### LINKED ISSUE #25793: [TC-WAKEONLAN-4.1]  Mac Address returned is Empty for wake-on-lan-mac-address attribute.
**Problem**
When running [TC_MC_4.1], Step 1, the Mac Address returned is Empty for wake-on-lan-mac-address attribute for the wakeonlan  cluster.
Expect to see the mac address of the device, in this case, the Linux device running chip-tv-app.

image

**Command used to build the TV-app**
`built the tv-app in linux using scripts/examples/gn_build_example.sh      examples/tv-app/linux/      examples/tv-app/linux/out/tv-app   chip_inet_config_enable_ipv4=false  chip_enable_wake_on_lan=true   `  

**GitHub hash of the SDK that was being used**
https://github.com/project-chip/connectedhomeip/commit/2767b518fe59769720e848a9971f34838770da59 - Master

**PFA Log below**
image

[TC-WAKEONLAN-4.1.txt](https://github.com/project-chip/connectedhomeip/files/11047681/TC-WAKEONLAN-4.1.txt)

### PR DESCRIPTION
**Pull request by chrisdecenzo**
Fixes #25793 

Read Platform's default interface MAC address in tv-info dummy implementation
This can be changed when ported to other hardware

### CODE REVIEW

File: examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp [hunk 1]
```diff
@@ -19,24 +19,41 @@
...
+
+    uint8_t j = 0;
+    for (uint8_t i = 0; i < sizeof(macBuffer); i++)
+    {
+        j         = i * 2;
+        macStr[j] = (((macBuffer[i] & 0xF0) >> 4) & 0xF);
+        macStr[j] += (macStr[j] <= 9) ? '0' : ('A' - 10);
+        j++;
+        macStr[j] = (macBuffer[i] & 0x0F);
+        macStr[j] += (macStr[j] <= 9) ? '0' : ('A' - 10);
```
Comment (bzbarsky-apple) [inline examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp:48]: We have existing BytesToHex helpers, no?

Comment (chrisdecenzo) [reply examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp]: Updated to use BytesToHex helper function - thanks!

**Review by lazarkov — APPROVED** [review]

File: examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp [hunk 2]
```diff
@@ -19,24 +19,37 @@
...
 
 std::string getMacAddress()
 {
-    std::ifstream input("/sys/class/net/eth0/address");
-    std::string line;
-    std::getline(input, line);
-    return line;
+    uint8_t macBuffer[chip::DeviceLayer::ConfigurationManager::kPrimaryMACAddressLength];
+    MutableByteSpan mac(macBuffer);
+    if (chip::DeviceLayer::ConfigurationMgr().GetPrimaryMACAddress(mac) != CHIP_NO_ERROR)
```
Comment (jmeg-sfy) [inline examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp:33]: i know this is the reference demo app. 
but here we obtain no matter what the primary mac address (which may be Wi-Fi).
How do we handle a device ETH controller which don't have this WOL feature ?
Does it exist an interface api to know if this capability is supported ?
Are we assuming the device manufacturer to set nicely the PICS related to this feature ?

**Review by jmeg-sfy — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp (hunk 2 above)
Comment (chrisdecenzo) [reply examples/tv-app/linux/include/wake-on-lan/WakeOnLanManager.cpp:33]: this is an optional cluster so a device that does not support WoL would not include this cluster. Yes, the device maker needs to make sure that they include (or don't include) this cluster. Also, they need to make sure that the Mac address used is for the correct network interface

**Review by andy31415 — APPROVED** [review]

**Pull request merged**